annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,283 @@
+                                    .find(input = folder)!!.groupValues[1]
+                            if (user.isBlank() || eventType.isBlank()) {
+                                throw RuntimeException(
+                                        "Missing annotation information (user/subtype) for $it")
+                            }
+                            val documentName: String = it.name
+                            // TODO: count events rather than annotated documents
+                            // https://github.com/isi-vista/curated-training-annotator/issues/41
+                            val jsonString: String = it.readText()
+                            val regex = Regex(pattern = """"CTEventSpan" : \[""")
+                            if (jsonString.contains(regex)) {
+                                docsList.add(DocumentAnnotation(documentName, user, eventType))
+                            }
+                        }
+                    }
+        }
+
+        /**
+         * Find all existing JSON files in the output directory and return
+         * the most recent file; else return null
+         */
+        fun locatePreviousStats(statsDir: File): StatsReport? {
+            val jsonReports: MutableList<StatsReport> = mutableListOf()
+            statsDir.walk().forEach {
+                if (it.name.endsWith(".json")) {
+                    val datePattern = Regex(pattern = """\d{4}-\d{2}-\d{2}""")
+                    val fileDate = datePattern
+                            .find(input = it.name)?.value
+                    if (fileDate != null) {
+                        jsonReports.add(StatsReport(it, fileDate))
+                    } else {
+                        logger.debug("No date found for JSON file ${it.name}")
+                    }
+                }
+            }
+            return if (jsonReports.isEmpty()) {
+                // No previous JSON report found
+                null
+            } else {
+                val sortedJSON: List<StatsReport> = jsonReports.sortedBy {it.report.lastModified()}
+                sortedJSON.last()
+            }
+        }
+
+        /**
+         * Load the previous annotation statistics from the
+         * most recently modified JSON file (identified from `locatePreviousStats`)
+         */
+        fun loadPreviousStats(statsFile: File): AnnotationStats {
+            val mapper = jacksonObjectMapper()
+            return mapper.readValue<AnnotationStats>(statsFile)
+        }
+
+        /**
+         * Use the numbers in the new statistics and previous ones to get the differences
+         */
+        fun getDiffs(newStats: AnnotationStats, previousStats: AnnotationStats): AnnotationStats {
+            val totalDiff = newStats.total - previousStats.total
+            val userDiff: MutableMap<String, Int?> = mutableMapOf()
+            for (user in newStats.byUser.keys) {
+                if (previousStats.byUser[user] != null)
```

Comment:
Always use `{`s on your ifs: https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/

Reply:
Oops, that was a mistake